main/openrc/firstboot.initd

```diff
@@ -0,0 +1,44 @@
+#!/sbin/openrc-run
+
+# The first boot init service
+
+# read kernel options
+init_KOPT() {
+	for opt in $(cat /proc/cmdline 2>/dev/null); do
+		case "$opt" in
+			ssh_*=*)
+				eval "KOPT_${opt%%=*}='${opt#*=}'" ;;
+		esac
+	done
+}
+
+depend() {
+	keyword -vserver -lxc
+	init_KOPT
+	if [ -n "$KOPT_ssh_key" ] || [ -n "$KOPT_ssh_pass" ]; then
+		need sshd
+	fi
+}
+
+
+start() {
+	init_KOPT
+	local rc=0
+	ebegin "Starting ${RC_SVCNAME}"
+	if [ -n "$KOPT_ssh_key" ] && [ ! -f "/root/.ssh/authorized_keys" ]; then
+		einfo "Fetching ssh keys"
+		mkdir -p /root/.ssh
+		case "$KOPT_ssh_key" in
+			http://*|https://*|ftp://*|ftps://*)
+				wget -q "$KOPT_ssh_key" -O /root/.ssh/authorized_keys
+				rc=$?;;
+			*) echo "$KOPT_ssh_key" > /root/.ssh/authorized_keys;;
+		esac
+	fi
+	if [ -n "$KOPT_ssh_pass" ]; then
+		echo "root:$KOPT_ssh_pass" | /usr/sbin/chpasswd
+		SSHD_OPTS="-o PermitRootLogin=yes" rc-service sshd start
```

Comment:
Hmm on second thought, looks like `SSHD_OPTS` has been deprecated in favor of `command_args ` ?